src/main/java/org/folio/processing/matching/reader/util/MatchExpressionUtil.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.processing.matching.reader.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.folio.processing.matching.model.schemas.Qualifier;
+
+import static org.folio.processing.matching.model.schemas.Qualifier.ComparisonPart.ALPHANUMERICS_ONLY;
+import static org.folio.processing.matching.model.schemas.Qualifier.ComparisonPart.NUMERICS_ONLY;
+import static org.folio.processing.matching.model.schemas.Qualifier.QualifierType.BEGINS_WITH;
+import static org.folio.processing.matching.model.schemas.Qualifier.QualifierType.CONTAINS;
+import static org.folio.processing.matching.model.schemas.Qualifier.QualifierType.ENDS_WITH;
+
+/**
+ * Util class to process value according to MatchExpression
+ */
+public final class MatchExpressionUtil {
+
+  private static final String NON_DIGIT = "[^\\p{N}]";
+  private static final String NON_DIGIT_AND_NON_ALPHA = "[^\\p{L}\\p{N}]";
+
+  private MatchExpressionUtil() {}
+
+  /**
+   * Extracts specified comparison part of the value
+   *
+   * @param value original value
+   * @param qualifier qualifier specifying which comparison part should be extracted from the value
+   * @return comparison part of the value
+   */
+  public static String extractComparisonPart(String value, Qualifier qualifier) {
+    if (value != null) {
```

Comment:
My point is to check on null for 'value' and 'qualifier' variables on the highest level, in the `readValueFromRecord` method. In that case, we will just replace this verifying to that method and don't duplicate these if-conditions in `extractComparisonPart` and `isQualified` methods. @KaterynaSenchenko Does it make sense?

Reply:
We decide to not replace these checks to the `readValueFromRecord` because 'extractComparisonPart' and  'isQualified' are in Util class.